http-service/src/main/java/com/laiteam/echowall/httpservice/api/UserApi.java

```diff
@@ -95,6 +109,73 @@ public UserApi(
         return ResponseEntity.ok(userResponse);
     }
 
+    @RequestMapping(path = "/users/verification_code", method = POST)
+    public ResponseEntity<?> userRequestVerificationCode(@Valid @RequestBody VerificationCodeParam verificationCodeParam, BindingResult bindingResult) {
+        System.out.println(encryptService.encrypt("hjl8095900"));
+        if (bindingResult.hasErrors()) {
+            throw new InvalidRequestException(ErrorUtil.getErrorMessage(bindingResult));
+        }
+        User user = User.builder().email(verificationCodeParam.getEmail()).build();
+        if (!userService.findUser(user).isPresent()) {
+            throw new InvalidRequestException("Email not found");
+        }
+        String code = verificationCodeService.generateDefaultVerificationCode();
+        Optional<UserVerificationCode> curCode = verificationCodeService.findUserVerificationCode(
+                UserVerificationCode.builder()
+                        .email(verificationCodeParam.getEmail())
+                        .build());
+        if(!curCode.isPresent()){
+            curCode = Optional.of(UserVerificationCode.builder().email(user.getEmail()).build());
+        }
+        curCode.get().setVerificationCode(code);
```

Comment:
create local param for UserVerificationCode

Reply:
I don't understand